Sources/Storage/Storage.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  Storage.swift
+//  StatsdClient
+//
+//  Created by Khoi Lai on 10/8/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+protocol Storage {
+    associatedtype Item
```

Comment:
Because the Storage is designed for Generic Storage Protocol. Do you think we should make the Key become generic too?
Somethings like that
```
protocol Storage {
    associatedtype Item
    associatedtype Key
}
```

Reply:
Good idea, `Key` could be anything that's `Hashable`. We'll delegate the constraint to the concrete type.